posthog/models/feature_flag.py

```diff
@@ -311,12 +356,55 @@ def get_active_feature_flags(
 
 # Return feature flags with per-user overrides
 def get_overridden_feature_flags(
-    team_id: int, distinct_id: str, groups: Dict[GroupTypeName, str] = {},
+    team_id: int, distinct_id: str, groups: Dict[GroupTypeName, str] = {}, hash_key_override: Optional[str] = None
 ) -> Dict[str, Union[bool, str, None]]:
-    feature_flags = get_active_feature_flags(team_id, distinct_id, groups)
 
-    # Get a user's feature flag overrides from any distinct_id (not just the canonical one)
+    all_feature_flags = FeatureFlag.objects.filter(team_id=team_id, active=True, deleted=False).only(
+        "id", "team_id", "filters", "key", "rollout_percentage", "ensure_experience_continuity"
+    )
+
+    flags_have_experience_continuity_enabled = any(
+        feature_flag.ensure_experience_continuity for feature_flag in all_feature_flags
+    )
+
     person = PersonDistinctId.objects.filter(distinct_id=distinct_id, team_id=team_id).values_list("person_id")[:1]
+
+    if flags_have_experience_continuity_enabled:
+        try:
+            person_id = person[0][0]
+        except IndexError:
+            person_id = None
+
+        if person_id is None and hash_key_override is not None:
```

Comment:
NIT: Can we move `if hash_key_override` to a top level if and make the person_id if/else more obvious. Was going back and forth when looking at this sequence of checks